crypto/err/err.c

```diff
@@ -812,3 +812,26 @@ int ERR_pop_to_mark(void)
     es->err_flags[es->top] &= ~ERR_FLAG_MARK;
     return 1;
 }
+
+int ERR_clear_last_mark(void)
+{
+    ERR_STATE *es;
+    int top;
+
+    es = ERR_get_state();
+    if (es == NULL)
+        return 0;
+
+    top = es->top;
+    while (es->bottom != top
+           && (es->err_flags[es->top] & ERR_FLAG_MARK) == 0) {
+        top -= 1;
+        if (top == -1)
+            top = ERR_NUM_ERRORS - 1;
```

Comment:
"(and elsewhere)" in my remark actually refers to ERR_pop_to_mark. Or in other words I find it hard to recognize circular buffer pattern in *both* cases. Note that I'm not saying that I *can't* recognize it, only find it hard to do so. It would be more appropriate to say that I find it harder than necessary. In other words I'd say that some alternative would be more readable...